packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+
+import { useCoreClerk, useCoreSession } from '../contexts';
+import { Col, Flex, Icon, Link, Text, useAppearance } from '../customizables';
+import { Portal } from '../elements/Portal';
+import { Eye } from '../icons';
+
+type EyeCircleProps = {
+  width: string;
+  height: string;
+  padding: string;
+};
+
+const EyeCircle: React.FC<EyeCircleProps> = ({ width, height, padding }) => {
+  return (
+    <Flex
+      id='impersonationEye'
+      sx={t => ({
+        padding,
+        transition: 'transform 800ms',
+        backgroundColor: t.colors.$danger500,
+        borderRadius: t.radii.$circle,
+      })}
+    >
+      <Icon
+        icon={Eye}
+        sx={{
+          width,
+          height,
+          transform: 'translateY(3.5px)',
+        }}
+      />
+    </Flex>
+  );
+};
+
+type FabContentProps = { id: string; paddingLeft: string; paddingRight: string };
+
+const FabContent: React.FC<FabContentProps> = ({ id, paddingLeft, paddingRight }) => {
+  const { signOut } = useCoreClerk();
+
+  return (
+    <Col
+      id='impersonationText'
+      sx={{
+        whiteSpace: 'nowrap',
```

Comment:
🔧 I'd also suggest to add `text-overflow: ellipsis;` and maybe `overflow: hidden;` so that the UI doesn't break when rendering the dynamic user identifier

Reply:
Done ✔️